### PR TITLE
[IMP] data_merge: merge company_dependent fields

### PR DESCRIPTION
**Pull request by fdamhaut**
Merge company_dependent fields when merginf base_partner

task-id: 3103035

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by MissingNoShiny — COMMENTED** [review]
LGTM, thanks for your work :slightly_smiling_face: 
I just have one small formatting suggestion

File: odoo/addons/base/wizard/base_partner_merge.py [hunk 1]
```diff
@@ -229,6 +229,24 @@ def update_records(model, src, field_model='model', field_id='res_id'):
...
+        UPDATE ir_property AS ___ip1
+        SET res_id = %(destination_id)s
+        WHERE res_id IN %(source_ids)s
+        AND NOT EXISTS (
+             SELECT 1 
+             FROM ir_property AS ___ip2
+             WHERE ___ip2.res_id = %(destination_id)s
+             AND ___ip2.fields_id = ___ip1.fields_id
+             AND ___ip2.company_id = ___ip1.company_id
+        )""", params)
```
[the remark this answers is not in the document]
Comment (MissingNoShiny) [reply odoo/addons/base/wizard/base_partner_merge.py:248]: okay, sorry :pensive:

[the remark this answers is not in the document]
Comment (sofiagvaladze) [reply odoo/addons/base/wizard/base_partner_merge.py:248]: @fdamhaut if we pretty ask? :innocent: 

Agree on harder to write part, but it's more readable.

**Comment by sofiagvaladze** [comment]
@fdamhaut can you rebase and push so the branch is all green? Let's see if there is still the warning in logs, otherwise we will have to investigate it :bulb: 

Thanks ^^

**Review by xmo-odoo — COMMENTED** [review]
There is a typo in the commit message (merginf).

File: odoo/addons/base/wizard/base_partner_merge.py [hunk 2]
```diff
@@ -229,6 +229,24 @@ def update_records(model, src, field_model='model', field_id='res_id'):
...
         self.env.flush_all()
 
+        # Company-dependent fields
+        with self._cr.savepoint():
+            params = {
+                'destination_id': f'res.partner,{dst_partner.id}',
+                'source_ids': tuple(f'res.partner,{src}' for src in src_partners.ids),
+            }
+            self._cr.execute("""
+        UPDATE ir_property AS ___ip1
```
[the remark this answers is not in the document]
Comment (fdamhaut) [reply odoo/addons/base/wizard/base_partner_merge.py:239]: :shrug: I just took example on the earlier query in the file but It doesn't matter

**Comment by sofiagvaladze** [comment]
robodoo r+

**Comment by tfr-odoo** [comment]
> LGTM, thanks for your work 🙂 I just have one small formatting suggestion

You missed the elephant in the room, the res_id should not check only for res.partner,ID but for destination._name,ID

**Pull request closed without merging**